modular_skyrat/modules/modular_implants/code/nifsofts/station_pass.dm

```diff
@@ -0,0 +1,191 @@
+/obj/item/disk/nifsoft_uploader/station_pass
+	name = "Automatic Apprasial"
+	loaded_nifsoft = /datum/nifsoft/station_pass
+
+//Modular Persistence variables for the station_pass NIFSoft
+/datum/modular_persistence
+	///What name is saved to the station pass NIFSoft?
+	var/station_pass_nifsoft_name
+	///What message is saved to the station pass NIFSoft?
+	var/station_pass_nifsoft_message
+
+/datum/nifsoft/station_pass
+	name = "Poem of Communal Souls"
+	program_desc = "The Poem of Communal Souls was the first commission the Altspace Coven ever took; a rare occasion for their involvement in NIFSoft development. This program was originally commissioned by a then-underground group of ravers as a sort of 'social contagion' for the purpose of spreading peace, love, unity, and respect. The software operates by allowing different users running it to ambiently share 'Verses' with each other, small portions of their unique nanomachine fields that carry user-set messages; sometimes actual poetry, short biographies, or simple hope to meet and bond with other NIF users. Each trade of nanomachine packets represents a physical memory of the user who traded it, some long-time 'Poets' surrounded with a dazzling rainbow of different past messages."
+	persistence = TRUE
+
+	///Is the NIFSoft transmitting data?
+	var/transmitting_data = TRUE
+	///Is the NIFSoft recieving data?
+	var/recieving_data = TRUE
+	///What username is being sent out?
+	var/transmitted_name = ""
+	///What message is being sent to other users?
+	var/transmitted_message = ""
+	///What ckey is being used by the owner? This is mostly here so that messages can't get spammed
+	var/transmitted_identifier = ""
+
+	///What messages has the user recieved?
+	var/list/message_list = list()
+	///The datum that is being used to receive messages
+	var/datum/proximity_monitor/advanced/station_pass/proximity_datum
+
+/datum/nifsoft/station_pass/New()
+	. = ..()
+
+	if(!transmitted_name)
+		transmitted_name = linked_mob.name
+	if(!transmitted_message)
+		transmitted_message = "Hello, I am [transmitted_name], it's nice to meet you!"
+	transmitted_identifier = linked_mob.ckey
+
+	add_message("station_pass_nifsoft", name, "Hello World")
+	proximity_datum = new(linked_mob, 1)
+	proximity_datum.parent_nifsoft = src
+
+/datum/nifsoft/station_pass/New()
+	qdel(proximity_datum)
+	proximity_datum = null
+
+	return ..()
+
+///Adds a message to the message_list based off the name and message
+/datum/nifsoft/station_pass/proc/add_message(sender_identifier, recieved_name, recieved_message)
+	if(!recieved_message)
+		return FALSE
+	if(!name)
+		name = "Unknown User"
+	else
+		for(var/message in message_list)
+			if(message["identifier"] == sender_identifier)
+				message["sender_name"] = recieved_name
+				message["message"] = recieved_message
+				message["timestamp"] = station_time_timestamp()
+				return TRUE
```

Comment:
What's this meant to do? That feels like it's not actually adding a message, unless it's meant to only ever add one message per person? This could use a comment.